packages/api/src/controllers/asset.test.ts

```diff
@@ -316,5 +324,58 @@ describe("controllers/asset", () => {
         },
       });
     });
+    describe("chunked upload", () => {
+      jest.setTimeout(10000);
+      it("should start upload, stop it, resume it on tus test server", async () => {
+        const filename = "test.mp4";
+        const path = "/tmp";
+        const filePath = `${path}/${filename}`;
+        const res = await client.post("/asset/request-upload", {
+          name: "tus-test",
+        });
+        await createMockFile(filePath, 1024 * 1024 * 10);
+        const file = fs.createReadStream(filePath);
+        const { size } = fs.statSync(filePath);
+        expect(res.status).toBe(200);
+        let { tusEndpoint } = await res.json();
+        tusEndpoint = tusEndpoint.split("/api/")[1];
+        tusEndpoint = `${client.server.host}/api/${tusEndpoint}`;
+        await new Promise<void>((resolve, reject) => {
+          let aborted = false;
+          const upload = new tus.Upload(file, {
+            endpoint: tusEndpoint,
+            urlStorage: new (tus as any).FileUrlStorage("/tmp/metadata"),
+            metadata: {
+              filename,
+              filetype: "video/mp4",
+            },
+            uploadSize: size,
+            onError(error) {
+              reject(error);
+              throw error;
+            },
+            onProgress(bytesUploaded, bytesTotal) {
+              const percentage = parseFloat(
+                ((bytesUploaded / bytesTotal) * 100).toFixed(2)
+              );
+              if (!aborted && percentage > 0) {
+                async () => {
+                  await upload.abort(true);
+                  aborted = true;
+                };
+              }
+              async () => {
+                await resumeTestTusUpload(upload);
+              };
+            },
+            onSuccess() {
+              resolve();
+            },
+          });
+          upload.start();
+        });
+        fs.unlinkSync(filePath);
```

Comment:
You should try to use the `async` versions of these filesystem calls as much as possible. Here you're in an `async` function already, could just use `await fs.unlink(filePath)` instead.

Sync I/O is an anti-pattern in JavaScript since it only has 1 thread. If you block doing sync I/O everything else in the process will wait until that's done, and even in tests this could cause issues.